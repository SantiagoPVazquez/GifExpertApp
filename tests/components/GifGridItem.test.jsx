import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";


describe('Tests in <GifGridItem/>', () => { 


    const title = 'Ronin';
    const url = 'https://ronindev.com/ronin.jpg';
    
    test('should match the snapshot', () => { 
        
       const {container} =  render(<GifGridItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();

     });


    test('should show the image with the indicated URL and ALT', () => { 
        
        render(<GifGridItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
        
    });
    
    
    test('should show Title as a paragraph', () => { 
        
        render(<GifGridItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();

       });

 });