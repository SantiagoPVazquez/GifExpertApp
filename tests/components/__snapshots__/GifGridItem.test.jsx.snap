// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem/> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Ronin"
      src="https://ronindev.com/ronin.jpg"
    />
    <p>
      Ronin
    </p>
  </div>
</div>
`;
